src/widget/wlibrary.cpp

```diff
@@ -115,6 +118,10 @@ bool WLibrary::isTrackInCurrentView(const TrackId& trackId) {
 
 void WLibrary::slotSelectTrackInActiveTrackView(const TrackId& trackId) {
     //qDebug() << "WLibrary::slotSelectTrackInActiveTrackView" << trackId;
+    if (!trackId.isValid()) {
+        return;
+    }
```

Comment:
whats the intend behind this not being a `VERIFY_OR_DEBUG_ASSERT`?

Reply:
Slots at the boundary might be invoked with an empty/invalid value if no track is selected. All internal invocation only make sense with a valid track id.

Reply:
Makes sense. If the internal invocations are just sanity checks then, shouldn't they be less aggressive `DEBUG_ASSERT`s instead?

Reply:
Also, how would one know what member functions are at an external boundary and what not? They're all public by necessity so its difficult to differentiate.

Reply:
> Makes sense. If the internal invocations are just sanity checks then, shouldn't they be less aggressive `DEBUG_ASSERT`s instead?

No. The call should not proceed with an invalid value to prevent undefined behavior.

Reply:
> No. The call should not proceed with an invalid value to prevent undefined behavior.

I agree from a safety-centric perspective, but from what I can tell with my intermediate C and C++ experience is that inserting runtime checks into every level of the call stack (at least when its not at a module boundary) is somewhat frowned upon _because performance_. I assume you prefer to favor the safety over performance in this case?

Reply:
Performance is not an issue here. Verbose code is.

Reply:
Thank you for sharing your insight.